### DISCUSSION TITLE
Fetch Mysql  tables deletes formula columns!

### DISCUSSION THREAD

**Discussion by gmentzik** [Budibase/budibase#9874] [Help] [answered]
Hi there is a probable bug. I am on backend version 2.3.16 and app version 2.1.42. My app is using a Mysql database.
For some of my tables I am creating formula columns to express some relations. Eg. count related rows from another table. 
It seams that after performing fetch tables even without any Mysql schema change  at the specific table the formula columns and probably any column existing only in Budibase are lost!

Columns are not lost if you perform schema modifications through Budibase. Please check this out as can be at least very annoying and can break application if you rely on these extra columns.

**Comment by gmentzik** [comment, marked as answer]
Could be also related to this: https://github.com/Budibase/budibase/discussions/5539

**Reply by melohagan** [reply]
This may help: https://docs.budibase.com/docs/sql-datasource#fetching-specific-tables

But there is an open issue otherwise: https://github.com/Budibase/budibase/issues/9790

**Reply by gmentzik** [reply]
That's a temporary solution. Sorry for not reading the manual first. Thank you for mentioning! It would be nice though Budibase prevents such problems esp if there is no table alteration. For example now it doesn't loose booleans definition despite mysql having it as Tinyint. It would also be nice if there would be a warning on which and how tables will be affected/altered when Fetch tables action is applied.
